# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineCacheIRCompiler.h
# Commit: bcd63afc2d34
# Full Hash: bcd63afc2d3449830cfe6f51c03373f890ef6435
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2023-07-06 04:06:52
# Regressor Bug: 1837620
# File Overlap Count: 1
# Description:
#   Bug 1837620 - Part 6: Make edges for multiple shape guard weak too r=sfink
#   
#   To make the edges for the multiple shape guard weak we need a way to find and
#   update the objects used to hold these edges. The simplest way to do this is to
#   add a vector in the zone.  I don't believe that there are not huge numbers of
# ==============================================================================

diff -r 24e0b10be67a -r bcd63afc2d34 js/src/jit/BaselineCacheIRCompiler.h
--- a/js/src/jit/BaselineCacheIRCompiler.h	Wed Jul 05 16:58:38 2023 +0000
+++ b/js/src/jit/BaselineCacheIRCompiler.h	Wed Jul 05 16:58:38 2023 +0000
@@ -144,6 +144,19 @@
   CACHE_IR_COMPILER_UNSHARED_GENERATED
 };
 
+// Special object used for storing a list of shapes to guard against. These are
+// only used in the fields of CacheIR stubs and do not escape.
+class ShapeListObject : public ListObject {
+ public:
+  static const JSClass class_;
+  static const JSClassOps classOps_;
+  static ShapeListObject* create(JSContext* cx);
+  static void trace(JSTracer* trc, JSObject* obj);
+
+  Shape* get(uint32_t index);
+  bool traceWeak(JSTracer* trc);
+};
+
 }  // namespace jit
 }  // namespace js
 